### PR TITLE
Adding what's new entry for retentionConfig in mco

### PR DESCRIPTION
**Pull request by dockerymick**
https://github.com/open-cluster-management/backlog/issues/12628

### CODE REVIEW

**Comment by dockerymick** [comment]
@marcolan018 are there any properties that need to be added into the API file, based on this [doc](https://docs.google.com/document/d/1qejBPv2D1nx7JbsOXFSXOKfvnyirH0VR3vnhlJsXWz4/edit)?

**Comment by marcolan018** [comment]
@dockerymick 
there are more changes in the retention related configuration. It's under a new section named as "retentionConfig" under "spec", and some of them have the new default values like below

```
spec:
  retentionConfig:
    blockDuration: 2h
    cleanupInterval: 5m
    deleteDelay: 48h
    retentionInLocal: 24h
    retentionResolution1h: 0d
    retentionResolution5m: 180d
    retentionResolutionRaw: 30d
```

**Comment by marcolan018** [comment]
based on changes in https://github.com/open-cluster-management/multicluster-observability-operator/pull/553
the latest retention config should as below:

"spec", and some of them have the new default values like below

```
spec:
  advanced:
    retentionConfig:
      blockDuration: 2h
      cleanupInterval: 5m
      deleteDelay: 48h
      retentionInLocal: 24h
      retentionResolution1h: 0d
      retentionResolution5m: 180d
      retentionResolutionRaw: 30d
```

@clyang82 fyi

File: observability/observability_enable.adoc [hunk 1]
```diff
@@ -184,20 +184,21 @@ spec:
...
+  advanced:
+    retentionConfig:
+      blockDuration: 2h # Block duration for TSDB block
+      cleanupInterval: 5m # How often we should clean up partially uploaded blocks and blocks with deletion mark in the background when --wait has been enabled
+      deleteDelay: 48h # Time before a block marked for deletion is deleted from bucket
+      retentionInLocal: 24h # How long to retain raw samples on local storage
+      retentionResolution1h: 0d # How long to retain samples of 1 hour in bucket
+      retentionResolution5m: 180d # How long to retain samples of 5 minutes in bucket
+      retentionResolutionRaw: 30d # How long to retain raw samples in bucket
+    storageConfig: # Specifies the storage to be used by Observability
```
Comment (marcolan018) [inline observability/observability_enable.adoc:196]: for "storageConfig", it's still under "spec"
only "retentionConfig" is moved to "spec.advanced"

File: apis/observability.json.adoc [hunk 2]
```diff
@@ -104,11 +104,35 @@ __required__|Parameters describing the MultiClusterObservability resource to be
...
+        "blockDuration":
+        "cleanupInterval":
+        "deleteDelay":
+        "retentionInLocal":
+        "retentionResolution1h":
+        "retentionResolution5m":
+        "retentionResolutionRaw":
+        }
+      }
+      "storageConfig": {
```
Comment (dockerymick) [inline apis/observability.json.adoc:125]: Should this be in the advance section or spec?

**Pull request merged**